apps/builder/app/routes/rest.ai.op.ts

```diff
@@ -122,18 +122,30 @@ export const action = async ({ request }: ActionArgs) => {
       templateGenerator.createChain<GPTModelMessageFormat>();
 
     const results = await Promise.all(
-      generateTemplatePrompts.map((operation, index) =>
+      generateTemplatePrompts.map((operation) =>
         generationChain({
           model: generationModel,
           context: {
             prompt: operation.llmPrompt,
             components,
           },
-        }).then((result) => [index, result] as const)
+        })
       )
     );
 
-    for (const [index, result] of results) {
+    await Promise.all(
+      results.map(async (result) => {
+        if (result.success === false) {
+          return;
+        }
+        await queryImagesAndMutateTemplate({
+          template: result.data,
+          accessKey: env.UNSPLASH_ACCESS_KEY ?? "",
```

Comment:
instead of having a default empty string we should not try to use this at all and potentially show error or warning